app/services/TwoStepVerificationThrottle.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import org.joda.time.DateTime
+import play.api.Play.{current, _}
+
+trait TwoStepVerificationThrottle {
+  def hourlyLimit: HourlyLimit
+
+  def registrationMandatory(discriminator: String) = {
+    val userValue = Math.abs(discriminator.hashCode % 100)
```

Comment:
I am wondering how likely the last 2 digits are random and providing a balance spread based on a scale 0-99 considering the impementation is: http://docs.oracle.com/javase/6/docs/api/java/lang/String.html#hashCode()